aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,172 @@
+1. Blazor uses JavaScript to bootstrap the .NET runtime and configures the runtime to load the app's assemblies and required base class libraries (BCLs). Document object model (DOM) manipulation and browser API calls are handled via JavaScript interoperability (interop).
+
+## Compilation modes
+
+Mono aims run under WebAssembly in two modes: interpreted and ahead-of-time (AOT).
+
+### Interpreted mode
+
+In interpreted mode, the Mono runtime is compiled into WebAssembly bytecode. However, the .NET assembly files aren't compiled into bytecode. The browser loads and executes the Mono runtime. Mono then loads and executes standard .NET assemblies (*\*.dll* files) built by the normal .NET compilation toolchain.
+
+![Interpreted mode](index/_static/interpretedmode.png)
+
+Interpreted mode is similar to how the core internals of the desktop Common Language Runtime (CLR) work. The desktop CLR is distributed precompiled into native code, which then loads and executes .NET assembly files. One key difference is that the desktop CLR uses just-in-time (JIT) compilation extensively to make execution faster. Mono on WebAssembly is closer to a pure interpretation model.
+
+### Ahead-of-time (AOT) compilation mode
+
+In ahead-of-time (AOT) compilation mode, the app's .NET assemblies are transformed into WebAssembly binaries at build time. At runtime, there's no interpretation. The code executes immediately. It's still necessary to load part of the Mono runtime. Features for low-level .NET services are loaded, such as garbage collection. Features for parsing .NET assemblies aren't required and aren't loaded.
+
+![AOT mode](index/_static/aotmode.png)
+
+This is similar to how the [Ngen.exe tool](/dotnet/framework/tools/ngen-exe-native-image-generator) has historically allowed AOT compilation of .NET binaries into native machine code. More recently, CoreRT provides a complete native AOT .NET runtime for executing .NET binaries.
+
+## Interpreted versus AOT
+
+Interpreted mode provides a faster development cycle than AOT. When code is changed under the interpreted model, the app can be rebuilt and reloaded in the browser within a few seconds. An AOT rebuild might take several minutes to compile and reload. Because interpreted mode offers faster recompilation, it may be preferred for development. It remains an open question how to best apply interpreted and AOT modes of compilation in Blazor.
+
+## Browsers that don't support WebAssembly
+
+When the .NET runtime is compiled into WebAssembly, it's also compiled into [asm.js](https://en.wikipedia.org/wiki/Asm.js). *asm.js* is a subset of JavaScript and can be executed by JavaScript runtimes in browsers going back several years. When Blazor loads in the browser, it checks for WebAssembly support. If WebAssembly isn't supported, the *asm.js* runtime is loaded. *asm.js* isn't always used becuase it's larger and slower than the WebAssembly runtime.
```

Comment:
becuase -> because

Reply:
You don't have to compile the .NET runtime to asm.js yourself. Can we make it clearer that we provide the asm.js based implementation of the .NET runtime for you?